diff -up ImageMagick-7.0.11-2/configure.ac.3~ ImageMagick-7.0.11-2/configure.ac
--- ImageMagick-7.0.11-2/configure.ac.3~	2021-03-04 14:14:59.144464239 +0100
+++ ImageMagick-7.0.11-2/configure.ac	2021-03-04 14:15:56.113596235 +0100
@@ -1263,12 +1263,12 @@ if test "$enable_openmp" != 'no'; then
     # Clang (passes for GCC but uses different OpenMP implementation)
     if test "x$LIB_OMP" = x ; then
       if $CC --version 2>&1 | grep clang > /dev/null ; then
-        AC_CHECK_LIB([omp],[GOMP_parallel_start],[LIB_OMP="-lomp"],[],[])
+        AC_CHECK_LIB([omp],[GOMP_parallel_start],[LIB_OMP="-fopenmp"],[],[])
       fi
     fi
     # GCC
     if test "x$GOMP_LIBS" = x ; then
-      AC_CHECK_LIB([gomp],[GOMP_parallel_start],[GOMP_LIBS="-lgomp"],[],[])
+      AC_CHECK_LIB([gomp],[GOMP_parallel_start],[GOMP_LIBS="-fopenmp"],[],[])
     fi
   else
     # Sun CC
