diff -up ImageMagick-7.0.11-2/configure.ac.5~ ImageMagick-7.0.11-2/configure.ac
--- ImageMagick-7.0.11-2/configure.ac.5~	2021-03-04 14:16:10.723630088 +0100
+++ ImageMagick-7.0.11-2/configure.ac	2021-03-04 14:16:51.897725486 +0100
@@ -1929,7 +1929,7 @@ if test "$with_fpx" != 'no'; then
             AC_MSG_RESULT([no -- some components failed test])
             have_fpx='no (failed tests)'
         else
-            FPX_LIBS='-lfpx'
+            FPX_LIBS='-lfpx -lstdc++'
             AC_DEFINE([FPX_DELEGATE],[1],[Define if you have FlashPIX library])
             AC_MSG_RESULT([yes])
             have_fpx='yes'
