diff -up ImageMagick-7.1.1-15/MagickCore/module.c.2~ ImageMagick-7.1.1-15/MagickCore/module.c
--- ImageMagick-7.1.1-15/MagickCore/module.c.2~	2023-07-30 13:49:14.000000000 +0200
+++ ImageMagick-7.1.1-15/MagickCore/module.c	2023-09-15 14:56:21.854095406 +0200
@@ -66,18 +66,14 @@
 #include "MagickCore/utility.h"
 #include "MagickCore/utility-private.h"
 #if defined(MAGICKCORE_MODULES_SUPPORT)
-#if defined(MAGICKCORE_LTDL_DELEGATE)
-#include "ltdl.h"
-typedef lt_dlhandle ModuleHandle;
-#else
+#include <dlfcn.h>
 typedef void *ModuleHandle;
-#endif
 
 /*
   Define declarations.
 */
 #if defined(MAGICKCORE_LTDL_DELEGATE)
-#  define ModuleGlobExpression "*.la"
+#  define ModuleGlobExpression "*.so"
 #else
 #  if defined(_DEBUG)
 #    define ModuleGlobExpression "IM_MOD_DB_*.dll"
@@ -893,11 +889,6 @@ static MagickBooleanType IsModuleTreeIns
           if (status == MagickFalse)
             ThrowFatalException(ResourceLimitFatalError,
               "MemoryAllocationFailed");
-#if defined(MAGICKCORE_LTDL_DELEGATE)
-          if (lt_dlinit() != 0)
-            ThrowFatalException(ModuleFatalError,
-              "UnableToInitializeModuleLoader");
-#endif
           module_list=splay_tree;
         }
       UnlockSemaphoreInfo(module_semaphore);
@@ -994,11 +985,11 @@ MagickExport MagickBooleanType InvokeDyn
   /*
     Open the module.
   */
-  handle=(ModuleHandle) lt_dlopen(path);
+  handle=(ModuleHandle) dlopen(path, RTLD_NOW|RTLD_GLOBAL);
   if (handle == (ModuleHandle) NULL)
     {
       (void) ThrowMagickException(exception,GetMagickModule(),ModuleError,
-        "UnableToLoadModule","'%s': %s",name,lt_dlerror());
+        "UnableToLoadModule","'%s': %s",name,dlerror());
       return(MagickFalse);
     }
   /*
@@ -1014,10 +1005,10 @@ MagickExport MagickBooleanType InvokeDyn
     Execute the module.
   */
   ClearMagickException(exception);
-  image_filter=(ImageFilterHandler *) lt_dlsym(handle,name);
+  image_filter=(ImageFilterHandler *) dlsym(handle,name);
   if (image_filter == (ImageFilterHandler *) NULL)
     (void) ThrowMagickException(exception,GetMagickModule(),ModuleError,
-      "UnableToLoadModule","'%s': %s",name,lt_dlerror());
+      "UnableToLoadModule","'%s': %s",name,dlerror());
   else
     {
       size_t
@@ -1038,9 +1029,9 @@ MagickExport MagickBooleanType InvokeDyn
   /*
     Close the module.
   */
-  if (lt_dlclose(handle) != 0)
+  if (dlclose(handle) != 0)
     (void) ThrowMagickException(exception,GetMagickModule(),ModuleWarning,
-      "UnableToCloseModule","'%s': %s",name,lt_dlerror());
+      "UnableToCloseModule","'%s': %s",name,dlerror());
   return(exception->severity < ErrorException ? MagickTrue : MagickFalse);
 }
 
@@ -1285,11 +1276,11 @@ MagickPrivate MagickBooleanType OpenModu
   */
   (void) LogMagickEvent(ModuleEvent,GetMagickModule(),
     "Opening module at path \"%s\"",path);
-  handle=(ModuleHandle) lt_dlopen(path);
+  handle=(ModuleHandle) dlopen(path, RTLD_NOW|RTLD_GLOBAL);
   if (handle == (ModuleHandle) NULL)
     {
       (void) ThrowMagickException(exception,GetMagickModule(),ModuleError,
-        "UnableToLoadModule","'%s': %s",path,lt_dlerror());
+        "UnableToLoadModule","'%s': %s",path,dlerror());
       return(MagickFalse);
     }
   /*
@@ -1303,11 +1294,11 @@ MagickPrivate MagickBooleanType OpenModu
     Define RegisterFORMATImage method.
   */
   TagToModuleName(module_name,"Register%sImage",name);
-  module_info->register_module=(size_t (*)(void)) lt_dlsym(handle,name);
+  module_info->register_module=(size_t (*)(void)) dlsym(handle,name);
   if (module_info->register_module == (size_t (*)(void)) NULL)
     {
       (void) ThrowMagickException(exception,GetMagickModule(),ModuleError,
-        "UnableToRegisterImageFormat","'%s': %s",module_name,lt_dlerror());
+        "UnableToRegisterImageFormat","'%s': %s",module_name,dlerror());
       return(MagickFalse);
     }
   (void) LogMagickEvent(ModuleEvent,GetMagickModule(),
@@ -1317,11 +1308,11 @@ MagickPrivate MagickBooleanType OpenModu
     Define UnregisterFORMATImage method.
   */
   TagToModuleName(module_name,"Unregister%sImage",name);
-  module_info->unregister_module=(void (*)(void)) lt_dlsym(handle,name);
+  module_info->unregister_module=(void (*)(void)) dlsym(handle,name);
   if (module_info->unregister_module == (void (*)(void)) NULL)
     {
       (void) ThrowMagickException(exception,GetMagickModule(),ModuleError,
-        "UnableToRegisterImageFormat","'%s': %s",module_name,lt_dlerror());
+        "UnableToRegisterImageFormat","'%s': %s",module_name,dlerror());
       return(MagickFalse);
     }
   (void) LogMagickEvent(ModuleEvent,GetMagickModule(),
@@ -1473,7 +1464,7 @@ static void TagToCoderModuleName(const c
   if (IsEventLogging() != MagickFalse)
     (void) LogMagickEvent(TraceEvent,GetMagickModule(),"%s",tag);
 #if defined(MAGICKCORE_LTDL_DELEGATE)
-  (void) FormatLocaleString(name,MagickPathExtent,"%s.la",tag);
+  (void) FormatLocaleString(name,MagickPathExtent,"%s.so",tag);
   (void) LocaleLower(name);
 #else
 #if defined(MAGICKCORE_WINDOWS_SUPPORT)
@@ -1527,7 +1518,7 @@ static void TagToFilterModuleName(const
 #elif !defined(MAGICKCORE_LTDL_DELEGATE)
   (void) FormatLocaleString(name,MagickPathExtent,"%s.dll",tag);
 #else
-  (void) FormatLocaleString(name,MagickPathExtent,"%s.la",tag);
+  (void) FormatLocaleString(name,MagickPathExtent,"%s.so",tag);
 #endif
 }
 
@@ -1624,10 +1615,10 @@ static MagickBooleanType UnregisterModul
   if (module_info->unregister_module == NULL)
     return(MagickTrue);
   module_info->unregister_module();
-  if (lt_dlclose((ModuleHandle) module_info->handle) != 0)
+  if (dlclose((ModuleHandle) module_info->handle) != 0)
     {
       (void) ThrowMagickException(exception,GetMagickModule(),ModuleWarning,
-        "UnableToCloseModule","'%s': %s",module_info->tag,lt_dlerror());
+        "UnableToCloseModule","'%s': %s",module_info->tag,dlerror());
       return(MagickFalse);
     }
   return(MagickTrue);
diff -up ImageMagick-7.1.1-15/configure.ac.2~ ImageMagick-7.1.1-15/configure.ac
--- ImageMagick-7.1.1-15/configure.ac.2~	2023-09-15 14:55:08.569658682 +0200
+++ ImageMagick-7.1.1-15/configure.ac	2023-09-15 14:55:41.532340987 +0200
@@ -1694,7 +1694,7 @@ if test "$with_ltdl" = 'yes'; then
       AC_MSG_RESULT([no -- some components failed test])
       have_ltdl='no (failed tests)'
     else
-      LTDL_LIBS='-lltdl'
+      LTDL_LIBS='-ldl'
       LIBS="$LTDL_LIBS $LIBS"
       AC_DEFINE([LTDL_DELEGATE],[1],[Define if using libltdl to support dynamically loadable modules and OpenCL])
       AC_DEFINE_UNQUOTED([LTDL_MODULE_EXT],["${shrext_cmds}"],[Native module suffix])
